src/petals/server/handler.py

```diff
@@ -47,6 +51,8 @@ def __init__(
         dht: DHT,
         module_backends: Dict[str, TransformerBackend],
         *,
+        push_manager: SyncManager,
+        session_pipes: Dict[str, Tuple[PipeConnection, threading.Lock]],
```

Comment:
`rpc_push()` may be received by a connection handler different from the one holding the inference session, so we use some multiprocess communication here.